src/ui/authenticationInApp/ViewAuthenticationVerificationSessionByEmailNeeded.qml

```diff
@@ -27,40 +30,84 @@ Item {
+        VPNAuthInApp.reset();
     }
+    _menuButtonAccessibleName: "Back"
+    _headlineText: "Enter verification code"
+    _subtitleText: "Open your email and enter the verification code it that was sent."
+    _imgSource: "qrc:/nebula/resources/verification-code.svg"
+
+    _inputs: ColumnLayout {
+        spacing: VPNTheme.theme.vSpacingSmall
+
+        VPNBoldLabel {
+            id: codeLabel
+            text: "Verification code"
+        }
+
+        VPNTextField {
+            id: codeInput
+            hasError: false
+            _placeholderText: "Enter 6-digit code"
+            Layout.fillWidth: true
+        }
+
+        VPNContextualAlerts {
+            id: passwordInputCreateWarnings
+            Layout.fillWidth: true
 
-    VPNTextField {
-        id: codeInput
+            messages: [
+                {
+                    type: "error",
+                    message: "Invalid code entry",
+                    visible: !createAccountButton.enabled
+                }
+            ]
+        }
 
-        anchors.top: msg.bottom
-        anchors.bottomMargin: 24
-        width: parent.width
+        VPNButton {
+            id: createAccountButton
+            enabled: codeInput.text && codeInput.text.length === VPNAuthInApp.getVerificationCodeLength()
+            text: "Verify" + VPNAuthInApp.getVerificationCodeLength()
+            Layout.fillWidth: true
+
+            onClicked: {
+                if (enabled) {
+                    VPNAuthInApp.verifySessionEmailCode(codeInput.text);
+                }
+            }
+        }
     }
 
-    VPNButton {
-        id: codeButton
-        anchors.top: codeInput.bottom
-        anchors.bottomMargin: 24
-        text: "Verify" // TODO
-        anchors.horizontalCenterOffset: 0
-        anchors.horizontalCenter: parent.horizontalCenter
-        radius: 5
-        onClicked: VPNAuthInApp.verifySessionEmailCode(codeInput.text);
+    _footerContent: Column {
+        Layout.alignment: Qt.AlignHCenter
+        spacing: VPNTheme.theme.windowMargin
+
+        VPNLinkButton {
+            labelText: "Resend code"
+            anchors.horizontalCenter: parent.horizontalCenter
+            onClicked: {
+                VPNAuthInApp.resendVerificationSessionCodeEmail();
+            }
+        }
+
+        VPNLinkButton {
+            labelText: "Cancel"
+            fontName: VPNTheme.theme.fontBoldFamily
+            anchors.horizontalCenter: parent.horizontalCenter
+            linkColor: VPNTheme.theme.redButton
+            onClicked: VPNAuthInApp.reset()
+        }
+
     }
 
-    VPNButton {
-        anchors.top: codeButton.bottom
-        anchors.bottomMargin: 24
-        text: "Send new code" // TODO
-        anchors.horizontalCenterOffset: 0
-        anchors.horizontalCenter: parent.horizontalCenter
-        radius: 5
-        onClicked: VPNAuthInApp.resendVerificationSessionCodeEmail();
+    Component.onCompleted: {
+        console.log("SESSION VERIFICATION BY EMAIL");
```

Comment:
When ready, we can remove these messages.

Reply:
Done!